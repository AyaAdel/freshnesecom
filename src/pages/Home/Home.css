.home-main {
  display: flex;
  padding: 30px 45px;
}

.home-main aside {
  flex-basis: 269px;
  margin-right: 32px;
  flex-shrink: 0;
}
