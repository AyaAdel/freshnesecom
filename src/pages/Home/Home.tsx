import { useEffect } from 'react';
import Products from './Products/Products';
import ProductsCategories from './ProductsCategories/ProductsCategories';
import Header from 'pages/Home/Header/Header';
import Footer from 'pages/Home/Footer/Footer';
import { getAllProducts, allProductsCategories } from 'services/products';
import { useAppDispatch } from 'app/hooks';
import type { AppDispatch } from 'app/store';
import { loadProducts } from 'features/products/productsSlice';

import { initCategories } from 'features/productsCategories/productsCategoriesSlice';

import './Home.css';

export default function Home() {
  const dispatch: AppDispatch = useAppDispatch();

  useEffect(() => {
    getAllProducts().then((response) => dispatch(loadProducts(response)));
    allProductsCategories().then((response) =>
      dispatch(initCategories(response))
    );
  }, []);

  return (
    <>
      <Header />
      <main className="home-main">
        <aside>
          <ProductsCategories />
        </aside>
        <Products />
      </main>
      <Footer />
    </>
  );
}
