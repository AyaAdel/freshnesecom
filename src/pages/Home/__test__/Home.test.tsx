import React from 'react';
import renderer from 'react-test-renderer';
import { RenderComponent } from 'utils/test/render';
import Home from '../Home';

function MyComponent() {
  return (
    <RenderComponent>
      <Home />
    </RenderComponent>
  );
}

it('Render Home Correctly', () => {
  const tree = renderer.create(<MyComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
