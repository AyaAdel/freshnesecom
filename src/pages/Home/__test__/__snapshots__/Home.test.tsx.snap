// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home Correctly 1`] = `
Array [
  <div
    className="header-info"
  >
    <div
      className="header-info__left-side"
    >
      <a
        href="#"
      >
        Chat with us
      </a>
      <span>
        +420 336 775 664
      </span>
      <span>
        info@freshnesecom.com
      </span>
    </div>
    <div
      className="header-info__right-side"
    >
      <a
        href="#"
      >
        Blog
      </a>
      <a
        href="#"
      >
        About Us
      </a>
      <a
        href="#"
      >
        Careers
      </a>
    </div>
  </div>,
  <div
    className="header"
  >
    <img
      src="brand.svg"
    />
    <div
      className="search-products"
    >
      <div
        className="dropdown-list"
        data-testid="dropdown-list"
      >
        <div
          className="dropdown-list__list__selected"
          onClick={[Function]}
        >
          <span
            data-testid="dropdown-list-selected"
          >
            All Categories
          </span>
          <img
            src="chevron-down.svg"
          />
        </div>
      </div>
      <input
        className="search-products__input"
        onChange={[Function]}
        placeholder="Search Products, categories ..."
        value=""
      />
      <img
        className="search-products__icon"
        src="search.svg"
      />
    </div>
    <div
      className="header__right-side"
    >
      <img
        className="icon"
        src="user.svg"
      />
      <div
        className="cart"
      >
        <img
          className="icon"
          src="cart.svg"
        />
      </div>
    </div>
  </div>,
  <nav
    className="navbar"
  >
    <div
      className="dropdown-list"
      data-testid="dropdown-list"
    >
      <div
        className="dropdown-list__list__selected"
        onClick={[Function]}
      >
        <span
          data-testid="dropdown-list-selected"
        >
          Bakery
        </span>
        <img
          src="chevron-down.svg"
        />
      </div>
    </div>
    <div
      className="dropdown-list"
      data-testid="dropdown-list"
    >
      <div
        className="dropdown-list__list__selected"
        onClick={[Function]}
      >
        <span
          data-testid="dropdown-list-selected"
        >
          Fruit and Vegetables
        </span>
        <img
          src="chevron-down.svg"
        />
      </div>
    </div>
    <div
      className="dropdown-list"
      data-testid="dropdown-list"
    >
      <div
        className="dropdown-list__list__selected"
        onClick={[Function]}
      >
        <span
          data-testid="dropdown-list-selected"
        >
          Meat and fish
        </span>
        <img
          src="chevron-down.svg"
        />
      </div>
    </div>
    <div
      className="dropdown-list"
      data-testid="dropdown-list"
    >
      <div
        className="dropdown-list__list__selected"
        onClick={[Function]}
      >
        <span
          data-testid="dropdown-list-selected"
        >
          Drinks
        </span>
        <img
          src="chevron-down.svg"
        />
      </div>
    </div>
    <div
      className="dropdown-list"
      data-testid="dropdown-list"
    >
      <div
        className="dropdown-list__list__selected"
        onClick={[Function]}
      >
        <span
          data-testid="dropdown-list-selected"
        >
          Kitchen
        </span>
        <img
          src="chevron-down.svg"
        />
      </div>
    </div>
    <div
      className="dropdown-list"
      data-testid="dropdown-list"
    >
      <div
        className="dropdown-list__list__selected"
        onClick={[Function]}
      >
        <span
          data-testid="dropdown-list-selected"
        >
          Special nutrition
        </span>
        <img
          src="chevron-down.svg"
        />
      </div>
    </div>
    <div
      className="dropdown-list"
      data-testid="dropdown-list"
    >
      <div
        className="dropdown-list__list__selected"
        onClick={[Function]}
      >
        <span
          data-testid="dropdown-list-selected"
        >
          Baby
        </span>
        <img
          src="chevron-down.svg"
        />
      </div>
    </div>
    <div
      className="dropdown-list"
      data-testid="dropdown-list"
    >
      <div
        className="dropdown-list__list__selected"
        onClick={[Function]}
      >
        <span
          data-testid="dropdown-list-selected"
        >
          pharmacy
        </span>
        <img
          src="chevron-down.svg"
        />
      </div>
    </div>
  </nav>,
  <main
    className="home-main"
  >
    <aside>
      <div
        className="list-group"
      >
        <h4
          className="list-group__title"
        >
          Categories
        </h4>
        <div
          className="list-group__links"
        />
      </div>
    </aside>
  </main>,
  <div
    className="footer"
  >
    <div
      className="footer__links"
    >
      <div
        className="list-group"
      >
        <h4
          className="list-group__title"
        >
          Get in touch
        </h4>
        <div
          className="list-group__links"
        >
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              About Us
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Careers
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Press Releases
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Blog
            </a>
          </div>
        </div>
      </div>
      <div
        className="list-group"
      >
        <h4
          className="list-group__title"
        >
          Connections
        </h4>
        <div
          className="list-group__links"
        >
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Facebook
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Twitter
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Instagram
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Youtube
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              LinkedIn
            </a>
          </div>
        </div>
      </div>
      <div
        className="list-group"
      >
        <h4
          className="list-group__title"
        >
          Earnings
        </h4>
        <div
          className="list-group__links"
        >
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Become an Affiliate
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Advertise your product
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Sell on Market
            </a>
          </div>
        </div>
      </div>
      <div
        className="list-group"
      >
        <h4
          className="list-group__title"
        >
          Account
        </h4>
        <div
          className="list-group__links"
        >
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Your account
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Returns Centre
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              100 % purchase protection
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Chat with us
            </a>
          </div>
          <div>
            <a
              className="list-group__link"
              href="#"
            >
              Help
            </a>
          </div>
        </div>
      </div>
    </div>
    <p
      className="footer__copyright"
    >
      Copyright Â© 2020 petrbilek.com
    </p>
  </div>,
]
`;
